scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -172,12 +172,61 @@
       (cc/write-compute-cluster conn {:compute-cluster/type :compute-cluster.type/kubernetes
                                       :compute-cluster/cluster-name compute-cluster-name}))))
 
+(defn get-bearer-token
+  [scoped-credentials]
+  (.refresh scoped-credentials)
+  (str "Bearer " (.getTokenValue (.getAccessToken scoped-credentials))))
+
+(def ^ScheduledExecutorService bearer-token-executor (Executors/newSingleThreadScheduledExecutor))
+
+(defn make-bearer-token-refresh-task
+  [api-client scoped-credentials]
+  (reify
+    Runnable
+    (run [_]
+      (try
+        (let [bearer-token (get-bearer-token scoped-credentials)]
+          (.setApiKey api-client bearer-token))
+        (catch Exception ex
```

Comment:
Should this retry, or should it suicide cook? If we don't refresh bearer token, in an hour or two, it'll stop functioning. But the critical diagnostic error message will be hours earlier. I want the error message to be fairly close to the end of the log. (So, e.g., setting a flag here and printing it around the authentication failure would also be acceptable.)

Reply:
Discussed offline